src/Mysqldump.php

```diff
@@ -837,47 +835,47 @@ private function listValues(string $tableName)
         $ignore = $this->settings->isEnabled('insert-ignore') ? '  IGNORE' : '';
         $count = 0;
 
+        $line = '';
         foreach ($resultSet as $row) {
             $count++;
             $values = $this->prepareColumnValues($tableName, $row);
             $valueList = implode(',', $values);
 
             if ($onlyOnce || !$this->settings->isEnabled('extended-insert')) {
                 if ($this->settings->isEnabled('complete-insert') && count($colNames)) {
-                    $lineSize += $this->write(sprintf(
+                    $line .= sprintf(
                         'INSERT%s INTO `%s` (%s) VALUES (%s)',
                         $ignore,
                         $tableName,
                         implode(', ', $colNames),
                         $valueList
-                    ));
-                } else {
-                    $lineSize += $this->write(
-                        sprintf('INSERT%s INTO `%s` VALUES (%s)', $ignore, $tableName, $valueList)
                     );
+                } else {
+                    $line .= sprintf('INSERT%s INTO `%s` VALUES (%s)', $ignore, $tableName, $valueList);
                 }
                 $onlyOnce = false;
             } else {
-                $lineSize += $this->write(sprintf(',(%s)', $valueList));
+                $line .= sprintf(',(%s)', $valueList);
             }
 
-            if (($lineSize > $this->settings->getNetBufferLength())
+            if ((strlen($line) > $this->settings->getNetBufferLength())
                 || !$this->settings->isEnabled('extended-insert')) {
                 $onlyOnce = true;
-                $lineSize = $this->write(';' . PHP_EOL);
+                $this->write($line . ';' . PHP_EOL);
```

Comment:
since we are buffering rows into batches with this PR, it might make sense to emit a event at this point in time, so a possible progress indicator could also update when a table dump is in-flight.

otherwise dumping huge tables will not see any progress updates for - in our case - 20-30 minutes.

see also https://github.com/druidfi/mysqldump-php/issues/39

I would propose something like

```suggestion
                $this->write($line . ';' . PHP_EOL);
                
                ($this->infoCallable)('table', ['name' => $tableName, 'rowProgress' => $count]);
```

Reply:
Another call before the foreach loop is also needed, otherwise the dump info will display an incorrect table name until the 1st million rows were processed.

Also maybe adding a state instead of rowProgress, e.g.:

Before the start of the foreach:
`['name' => $tableName, 'complete' => false, 'rowCount' => 0]`

Within the foreach loop:
`['name' => $tableName, 'complete' => false, 'rowCount' => $count]`

After the foreach loop:
`['name' => $tableName, 'complete' => true, 'rowCount' => $count]`

Reply:
valid points. I like your idea more than mine